test/extensions/filters/http/ext_authz/ext_authz_integration_test.cc

```diff
@@ -333,6 +349,19 @@ class ExtAuthzGrpcIntegrationTest : public Grpc::VersionedGrpcClientIntegrationP
     cleanup();
   }
 
+  void expectCheckRequestTimedout(bool timeout_on_check) {
+    setMeasureTimeoutOnCheckCreated(this->config_helper_, timeout_on_check);
+    initializeConfig(true);
+    setDownstreamProtocol(Http::CodecClient::Type::HTTP2);
+    HttpIntegrationTest::initialize();
+    initiateClientConnection(4);
+
+    response_->waitForEndStream();
+    EXPECT_TRUE(response_->complete());
+    EXPECT_EQ("403", response_->headers().getStatusValue());
```

Comment:
Is there a way to validate this happened via timeout?

Reply:
I can validate with EXPECT_LOG_CONTAINS

Reply:
A stat would be fairly easy to add I think to `stats_` and then make testing more robust.